### PR TITLE
Bulk Confirm

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
Confirming multiple users is currently time consuming and involves multiple steps. This PR adds two new endpoints, one for retrieving the public key of multiple organization users, and another for confirming multiple users.

### Code Changes
- **src/Api/Controllers/OrganizationUsersController.cs**: Add `confirm` and `public-keys` endpoints.
- **src/Core/Models/Api/Request/Organizations/OrganizationUserRequestModels.cs**: Add a new request model for BulkConfirm.
- **src/Core/Models/Api/Response/OrganizationUserResponseModel.cs**: Add two new response modewls, for returning the key and status of confirmation.
- **src/Core/Models/Data/OrganizationUserPublicKey.cs**: Representation of the response from the new sproc, `User_ReadPublicKeysByOrganizationUserIds`.
- **src/Core/Repositories/SqlServer/OrganizationUserRepository.cs**: Add function to get public keys for multiple organization users.
-  **src/Core/Repositories/SqlServer/UserRepository.cs**: Add a GetMany function to retrieve multiple users at a time.
- **src/Core/Services/Implementations/OrganizationService.cs**: Add bulk confirm. Refactored some common code.
- **src/Sql/dbo/Stored Procedures/User_ReadByIds.sql**: Add sproc for retrieving multiple users at a time.
- **src/Sql/dbo/Stored Procedures/User_ReadPublicKeysByOrganizationUserIds.sql**: Add sproc for retrieving public keys.

### Testing Considerations
We should do a regression sweep on ConfirmUser to verify it behaves the same.

### CODE REVIEW

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 1]
```diff
@@ -502,13 +505,187 @@ public async Task DeleteUsers_LastOwner(OrganizationUser orgUser, SutProvider<Or
...
+            var userService = Substitute.For<IUserService>();
+
+            org.PlanType = PlanType.EnterpriseAnnually;
+            orgUser.Status = OrganizationUserStatusType.Accepted;
+            orgUser.OrganizationId = confirmingUser.OrganizationId = org.Id;
+            orgUser.UserId = user.Id;
+            twoFactorPolicy.Type = PolicyType.TwoFactorAuthentication;
+            twoFactorPolicy.Enabled = true;
+            singleOrgPolicy.Type = PolicyType.SingleOrg;
+            singleOrgPolicy.Enabled = true;
```
Comment (addisonbeck) [inline test/Core.Test/Services/OrganizationServiceTests.cs:641]: This test looks ripe for a fixture customization attribute to abstract as much hard coded data manipulation out of the test as possible. `PlanType`, `orgUser.Status`, etc. are used in several tests, so I bet there are some lines that could get deleted from other tests too by creating this.

I see that there is a lot of data in these tests that is dependent on other parameters (`ou.UserId = user.Id` type stuff). There's not a pattern set up to implement these in our fixtures yet, and those should continue to be set in the test like you already have it.

**Review by addisonbeck — COMMENTED** [review]
giphy

File: src/Core/Services/IOrganizationService.cs [hunk 2]
```diff
@@ -33,17 +33,20 @@ public interface IOrganizationService
...
-        Task ResendInvitesAsync(Guid organizationId, Guid? invitingUserId, IEnumerable<Guid> organizationUsersId);
+        Task<IEnumerable<Tuple<OrganizationUser, string>>> ResendInvitesAsync(Guid organizationId, Guid? invitingUserId, IEnumerable<Guid> organizationUsersId);
         Task ResendInviteAsync(Guid organizationId, Guid? invitingUserId, Guid organizationUserId);
         Task<OrganizationUser> AcceptUserAsync(Guid organizationUserId, User user, string token,
             IUserService userService);
         Task<OrganizationUser> AcceptUserAsync(string orgIdentifier, User user, IUserService userService);
         Task<OrganizationUser> ConfirmUserAsync(Guid organizationId, Guid organizationUserId, string key,
             Guid confirmingUserId, IUserService userService);
+        Task<List<Tuple<OrganizationUser, string>>> ConfirmUsersAsync(Guid organizationId, Dictionary<Guid, string> keys,
+            Guid confirmingUserId, IUserService userService);
```
[the remark this answers is not in the document]
Comment (Hinton) [reply src/Core/Services/IOrganizationService.cs:44]: Cyclic dependency. `UserService` requires the `OrganizationService`.

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 1 above)
Comment (Hinton) [reply test/Core.Test/Services/OrganizationServiceTests.cs]: Added a few fixtures which should clean things up.

**Review by addisonbeck — DISMISSED** [review]

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**